app/main/controllers/activity/RTMediaActivity.php

```diff
@@ -53,71 +68,110 @@ function create_activity_html( $type = 'activity' ) {
+								%4$s
+							</h4>
+						</div>
+					</a>',
+					esc_url( get_rtmedia_permalink( $media->id ) ),
+					$this->media( $media ),
+					esc_attr( $media->media_title ),
+					$media->media_title
+				);
+			} elseif ( 'audio' === $media->media_type || 'video' === $media->media_type ) {
+				// Markup for audio and video media type with link only on media (title).
+				$media_content .= sprintf(
+					'<div class="rtmedia-item-thumbnail">
+						%1$s
+					</div>
+					<div class="rtmedia-item-title">
+						<h4 title="%2$s">
+							<a href="%3$s">
+								%4$s
+							</a>
+						</h4>
+					</div>',
+					$this->media( $media ),
+					esc_attr( $media->media_title ),
+					esc_url( get_rtmedia_permalink( $media->id ) ),
+					$media->media_title
+				);
+			} else {
+				// Markup for all the other media linke docs and other files where anchor tag the markup is comming from add-on itself.
+				$media_content .= sprintf(
+					'<div class="rtmedia-item-thumbnail">
+							%1$s
+					</div>
+					<div class="rtmedia-item-title">
+							<h4 title="%2$s">
+								%3$s
+							</h4>
+					</div>',
+					$this->media( $media ),
+					esc_attr( $media->media_title ),
+					$media->media_title
+				);
 			}
 
-			$li_content .= '</li>';
+			$media_content .= '</li>';
 			$count ++;
 		}
-
-		if( 'activity' == $type ){
-			$html .= '<ul class="rtmedia-list ' . esc_attr( $rtmedia_activity_ul_class ) . ' rtmedia-activity-media-length-' . esc_attr( $count ) . '">';
-		}else{
-			$html .= '<ul class="rtmedia-'.$type.'-list ' . esc_attr( $rtmedia_activity_ul_class ) . ' rtmedia-activity-media-length-' . esc_attr( $count ) . '">';
+		$media_container_start = '';
+		if ( 'activity' === $type ) {
+			$media_container_start .= sprintf(
+				'<ul class="rtmedia-list %1$s rtmedia-activity-media-length-%2$s">',
+				esc_attr( $rtmedia_activity_ul_class ),
+				esc_attr( $count )
+			);
+		} else {
+			$media_container_start .= sprintf(
+				'<ul class="rtmedia-%1$s-list %2$s rtmedia-activity-media-length-%3$s">',
+				esc_attr( $type ),
+				esc_attr( $rtmedia_activity_ul_class ),
+				esc_attr( $count )
+			);
 		}
-
-		$html .= $li_content;
-		$html .= '</ul>';
-		$html .= '</div>';
-
-		// Bypass comment links limit.
-		add_filter(
-			'option_comment_max_links',
-			function ( $values ) {
-				$rtmedia_attached_files = filter_input( INPUT_POST, 'rtMedia_attached_files', FILTER_DEFAULT, FILTER_REQUIRE_ARRAY );
-				// Check  if files available.
-				if ( is_array( $rtmedia_attached_files ) && ! empty( $rtmedia_attached_files[0] ) ) {
-					// One url of image and other for anchor tag.
-					$values = count( $rtmedia_attached_files ) * 2;
-				}
-				return $values;
-			}
-		);
-
-		return bp_activity_filter_kses( $html );
+		$media_container_end = '</ul>';
+
+		$media_list  = $media_container_start;
+		$media_list .= $media_content;
+		$media_list .= $media_container_end;
+
+		/**
+		 * Filters the output of the activity contents before save.
```

Comment:
before display

Reply:
No, this function modifies the content before saving it into the database, so I've add `before save`.